### PR TITLE
Create separate API route for each trpc router

### PR DESCRIPTION
**Pull request by keithwillcode**
## What does this PR do?

Separates all tRPC routers into their own subfolders so we only load the dependencies needed per router instead of for all of them.

**Environment**: Staging(main branch) / Production

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

Need to test basically the entire app.

### CODE REVIEW

File: apps/web/pages/api/trpc/appRoutingForms/[trpc].ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+   * Enable query batching
+   */
+  batching: {
+    enabled: true,
+  },
+  /**
+   * @link https://trpc.io/docs/caching#api-response-caching
+   */
+  responseMeta({ ctx, paths, type, errors }) {
```
Comment (keithwillcode) [inline apps/web/pages/api/trpc/appRoutingForms/[trpc].ts:36]: This needs to be moved to a common, reusable function.

File: packages/trpc/react/trpc.ts [hunk 2]
```diff
@@ -41,17 +69,45 @@ export const trpc = createTRPCNext<AppRouter, NextPageContext, "ExperimentalSusp
...
-          }),
+          true: (runtime) => {
+            const links = Object.fromEntries(
+              ENDPOINTS.map((endpoint) => [endpoint, httpLink({ url: url + "/" + endpoint })(runtime)])
+            );
+            return (ctx) => {
+              const parts = ctx.op.path.split(".");
+              let endpoint = '';
+              let path = '';
+              if (parts.length == 2) {
```
Comment (keithwillcode) [inline packages/trpc/react/trpc.ts:82]: This logic is gross. We should refactor the structure of the viewer, loggedInViewer and public routers to make this cleaner.

**Comment by keithwillcode** [comment]
This started out as a separate experiment for the public router but is truly an extension of #8041 so if we choose this path, I will cherry-pick commits over to that PR to build on @juliusmarminge work.

**Comment by keithwillcode** [comment]
Initial tests are showing that this does in fact make a massive different on cold starts. We didn't see the benefits in #8041 because it only separated the slots router out. In my initial follow up experiment, separating the public tRPC router to a subfolder resulted in a cold start load time of `public.viewer.i18n` of 2 seconds instead of 15+.

File: apps/web/pages/api/trpc/appRoutingForms/[trpc].ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    // Our cache can change depending on our current paths value. Since paths is an array,
+    // we want to create a map that can match potential paths with their desired cache value
+    const cacheRules = {
+      "viewer.public.session": `no-cache`,
+      "viewer.public.i18n": `no-cache`,
+      // Revalidation time here should be 1 second, per https://github.com/calcom/cal.com/pull/6823#issuecomment-1423215321
+      "viewer.public.slots.getSchedule": `no-cache`, // FIXME
+      "viewer.public.cityTimezones": `max-age=${ONE_DAY_IN_SECONDS}, stale-while-revalidate`,
+    } as const;
```
Comment (juliusmarminge) [inline apps/web/pages/api/trpc/appRoutingForms/[trpc].ts:72]: These paths are incorrect now

File: apps/web/pages/api/trpc/apps/[trpc].ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  },
+  /**
+   * @link https://trpc.io/docs/caching#api-response-caching
+   */
+  responseMeta({ ctx, paths, type, errors }) {
+
+    // Some helpers relevant to this function only
+    const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
+    // assuming you have all your public routes with the keyword `public` in them
+    const allPublic = paths && paths.every((path) => path.startsWith("viewer.public."));
```
Comment (juliusmarminge) [inline apps/web/pages/api/trpc/apps/[trpc].ts:41]: Same as above

File: apps/web/pages/api/trpc/auth/[trpc].ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  },
+  /**
+   * @link https://trpc.io/docs/caching#api-response-caching
+   */
+  responseMeta({ ctx, paths, type, errors }) {
+
+    // Some helpers relevant to this function only
+    const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
+    // assuming you have all your public routes with the keyword `public` in them
+    const allPublic = paths && paths.every((path) => path.startsWith("viewer.public."));
```
Comment (juliusmarminge) [inline apps/web/pages/api/trpc/auth/[trpc].ts:41]: ^^

File: apps/web/pages/api/trpc/availability/[trpc].ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  },
+  /**
+   * @link https://trpc.io/docs/caching#api-response-caching
+   */
+  responseMeta({ ctx, paths, type, errors }) {
+
+    // Some helpers relevant to this function only
+    const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
+    // assuming you have all your public routes with the keyword `public` in them
+    const allPublic = paths && paths.every((path) => path.startsWith("viewer.public."));
```
Comment (juliusmarminge) [inline apps/web/pages/api/trpc/availability/[trpc].ts:41]: ^^

File: apps/web/pages/api/trpc/appRoutingForms/[trpc].ts (hunk 3 above)
Comment (keithwillcode) [reply apps/web/pages/api/trpc/appRoutingForms/[trpc].ts:72]: Yeah, I'm waiting to update them when we refactor this method into 1 common place.

**Comment by keithwillcode** [comment]
Closing this because all work now going into #8041

**Pull request closed without merging**